agent/engine/task_manager_test.go

```diff
@@ -373,6 +557,18 @@ func TestContainerNextStateWithTransitionDependencies(t *testing.T) {
 			expectedTransitionActionable: false,
 			reason:                       dependencygraph.ErrContainerDependencyNotResolved,
 		},
+		// NONE -> RUNNING transition is not allowed and not actionable, when desired is Running and dependency is Restarting
```

Comment:
is NONE -> RESTARTING possible?

Reply:
This function test the next state of containers which are not in steady states. If `None` is the current status, and `Running` is the desired states, the next state of the container should be `Pulled` and then Agent will call Docker to pull image. Agent will never set a container desired status to be `Restarting`, so `None`-> `Restarting` will not be possible.